io-engine/src/core/logical_volume.rs

```diff
@@ -52,3 +54,90 @@ pub struct LvolSpaceUsage {
     /// clone.
     pub allocated_bytes_snapshot_from_clone: Option<u64>,
 }
+
+/// Arguments for replica creation.
+pub struct ReplicaArgs {
+    pub(crate) name: String,
+    pub(crate) size: u64,
+    pub(crate) uuid: String,
+    pub(crate) thin: bool,
+    pub(crate) entity_id: Option<String>,
+}
+
+/// This interface defines the high level operations which can be done on a
+/// pool. Pool-Specific details should be hidden away in the implementation as
+/// much as possible, though we can allow for extra pool specific options
+/// to be passed as parameters.
+#[async_trait::async_trait(?Send)]
+pub trait Pool {
```

Comment:
This is probably not the best location?

Reply:
> This is probably not the best location?

Yeah, doesn't seem to fit well in here with all things volume. Also, we should name it `PoolOps` as this is defining the operations only, and not the pool itself. On a separate note, this file is named logical_volume.rs - I assume that encompasses both the spdk lvol and lvm logical volume?

Reply:
what is your suggestion, move replicas&pool ops to another file (ex: pool.rs) or leave replica ops here and move pool only?
Initially it wasn't but I removed spdk specific bits a while back, so it can now be generic.

Reply:
Moving only pool ops sounds reasonable.